src/Gallery.js

```diff
@@ -58,16 +59,25 @@ class Gallery extends React.Component{
 	}
 	render(){
 		const resizedPhotos = this.scalePhotoDimensions();
-		style.margin = this.props.margin;
 		return(
 			<div id="Gallery" className="clearfix" ref={(c) => this._gallery = c}>
-				{resizedPhotos.map((photo,idx) => 
-		    		<div style={style} key={idx}>
-						<a href="#" onClick={(e) => this.props.onClickPhoto(idx, e)}>
-							<img src={photo.src} srcSet={photo.srcset.join()} sizes={photo.sizes.join()} style={{display:'block', border:0}} height={photo.height} width={photo.width} alt={photo.alt} />
-						</a>
-		    		</div>
-				)}
+				{resizedPhotos.map((photo,idx) => { 
+					let Image = (photo.component) ? photo.component : DefaultImage; 
+					return (
+						<Image
+							key={idx}
```

Comment:
Not that this is wrong. But personally I would set `key` to `photo.src`. It does prevent that photo's are showing up twice, but why would anyone do that anyway? Setting it to `photo.src` instead; results in better performance when dynamically adding / removing photo's. 

Setting it to `idx`, results in the best case to performance loss when adding an image on the beginning of the array. In the worst case; it causes some glitches. As `react` doesn't now that the new `idx === 3` is not the same as the last `idx === 3`, after an `photos.unshift(....)`.

Advice; change to `key={photo.key || photo.src}`. If the user needs to show the same photo multiple times, he can provide his own key's.